cpp/src/parquet/page_index.h

```diff
@@ -38,10 +41,10 @@ class RowGroupPageIndexReader;
 class PARQUET_EXPORT ColumnIndex {
  public:
   /// \brief Create a ColumnIndex from a serialized thrift message.
-  static std::unique_ptr<ColumnIndex> Make(const ColumnDescriptor& descr,
-                                           const void* serialized_index,
-                                           uint32_t index_len,
-                                           const ReaderProperties& properties);
+  static std::unique_ptr<ColumnIndex> Make(
+      const ColumnDescriptor& descr, const void* serialized_index, uint32_t index_len,
+      const ReaderProperties& properties,
+      const std::shared_ptr<Decryptor>& decryptor = NULLPTR);
```

Comment:
I think if we are sharing ownership the shared_ptr should be passed by value.  If we do not intend to share ownership pass by raw pointer.